### PR TITLE
Rc 25889 Terraform for GKE 

### PR DESCRIPTION
**Pull request by poornima-rafay**

### CODE REVIEW

File: examples/resources/rafay_gke_cluster/provider.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+terraform {
+  required_providers {
+    rafay = {
+      "source": "rafay/rafay",
+      "version": "1.1.13"
```
Comment (poornima-rafay) [inline examples/resources/rafay_gke_cluster/provider.tf:5]: @stephan-rafay should it say >= certain version? 
please confirm what this field should be

File: rafay/gke_flatten_util.go [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
+	}
+	var err error
+
+	var ret1 []interface{}
+	if in.Metadata != nil {
+		v, ok := obj["metadata"].([]interface{})
+		if !ok {
+			v = []interface{}{}
+		}
+		ret1 = flattenMetadataV3(in.Metadata, v)
```
Comment (MJjainam) [inline rafay/gke_flatten_util.go:40]: @poornima-rafay Out of curiousity..why can't we use flattenMetada function itself?

Comment (MJjainam) [reply rafay/gke_flatten_util.go:40]: Suggestion: we need not fetch 'v' and pass it to this function. It's anyway not being used.

**Review by chaitanya-rafay — APPROVED** [review]

File: rafay/gke_flatten_util.go (hunk 2 above)
Comment (poornima-rafay) [reply rafay/gke_flatten_util.go:40]: @MJjainam good question. I am not sure either. I am reusing the function that was written for AKS_V3. 
Also I see there's some difference between flattenMetaData() and flattenMetadataV3(). 
@thirumal-rafay do you know why we introduced flattenMetadataV3 for AKS?

File: examples/resources/rafay_gke_cluster/provider.tf (hunk 1 above)
Comment (stephan-rafay) [reply examples/resources/rafay_gke_cluster/provider.tf:5]:
```
terraform {
  required_providers {
    rafay = {
      version = ">= 1.1.15"
      source = "registry.terraform.io/RafaySystems/rafay"
    }
  }
}
```

**Pull request merged**